### PR TITLE
Return proper exceptions

### PR DESCRIPTION
**Pull request by yuzhu**

### CODE REVIEW

**Comment by yuzhu** [comment]
@calvinjia PTAL

File: experimental/table/server/master/src/main/java/alluxio/master/catalog/AlluxioCatalog.java [hunk 1]
```diff
@@ -139,7 +144,7 @@ public Table getTable(String dbName, String tableName) throws IOException {
   public List<String> getAllTables(String dbName) throws IOException {
     Database db = mDBs.get(dbName);
     if (db == null) {
-      throw new IOException(String.format("Database %s does not exist", dbName));
+      throw new NotFoundException(String.format("Database %s does not exist", dbName));
```
Comment (calvinjia) [inline experimental/table/server/master/src/main/java/alluxio/master/catalog/AlluxioCatalog.java]: Consider making a utility method for getting databases and tables by name if we need to do this logic often.

Comment (yuzhu) [reply experimental/table/server/master/src/main/java/alluxio/master/catalog/AlluxioCatalog.java]: done.

File: experimental/table/server/under/hive/src/main/java/alluxio/table/under/hive/HiveDatabase.java [hunk 2]
```diff
@@ -181,6 +183,8 @@ public UdbTable getTable(String tableName) throws IOException {
           HiveUtils.toProtoSchema(table.getAllCols()), tableUri.getPath(),
           Collections.singletonMap("unpartitioned", builder.build()),
           HiveUtils.toProto(table.getPartitionKeys()), partitions, table);
+    } catch (InvalidTableException e) {
+      throw new NotFoundException("Table " + tableName + " not found", e);
```
Comment (calvinjia) [inline experimental/table/server/under/hive/src/main/java/alluxio/table/under/hive/HiveDatabase.java]: Could we make the exception messages consistent as well as use `String.format` or just String concatenation consistently?

Comment (yuzhu) [reply experimental/table/server/under/hive/src/main/java/alluxio/table/under/hive/HiveDatabase.java]: fixed. using String concatenation only.

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Pull request merged**